lib/artifact.js

```diff
@@ -90,21 +90,41 @@ module.exports = (fs, logger, responseHandler, Metadata) => {
          * initialized with to a file. The user can pass in either a string
          * containing a path in their local file system, or a WriteStream to
          * pipe the data to. Returns the path the artifact was downloaded to.
+         * If a file path is passed in and an error response occurs it cleans
+         * up any residual file. We do not do this if a stream is passed in.
          *
          * @param {(string|WriteStream)} downloadLocation The local file system
          *                                              path or WriteStream to
          *                                              save the artifact to.
          * @return {Promise.<undefined>}
          */
         downloadToFile(downloadLocation) {
```

Comment:
So, I think the functionality should be no matter what inside this file we should clean up the file if it is a file. If they want to keep the file they can work around it by using `download`